desktop/logger/archiver.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+/**
+ * External dependencies
+ */
+const fs = require('fs');
+const JSZip = require('jszip');
+
+/**
+ * Internal dependencies
+ */
+const log = require('./')('desktop:lib:archiver');
+
+module.exports = {
+  /**
+   * Compresses `contents` to the archive at `dst`.
+   *
+   * @param {String[]} contents Paths to be zipped
+   * @param {String} dst Path to destination archive
+   * @param {function():void} onZipped Callback invoked when archive is complete
+   */
+  zipContents: (logPath, dst, onZipped) => {
+    const zip = new JSZip();
+    const contentPromise = new JSZip.external.Promise(function (
+      resolve,
+      reject
+    ) {
+      fs.readFile(logPath, function (err, data) {
+        if (err) {
+          log.warn('Unexpected error: ', err);
+          reject(err);
+        } else {
+          resolve(data);
+        }
+      });
+    });
```

Comment:
is this Promise necessary since our platform support already assumes a native `Promise`? (and arrow functions)

```js
zip.file( 'simplenote.log', new Promise( ( resolve, reject ) =>
	fs.readFile( logPath, ( error, data ) => {
		if (error) {
			log.warn('Unexpected error: ', error);
			reject(error);
		} else {
			resolve(data);
		}
} ) ) );
```

I suppose there could be some special interface that `JSZip.external.Promise` provides but I'm guessing they offer that as a compatibility layer for older Node/browser environments